FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatStorage.swift

```diff
@@ -51,20 +51,43 @@ final class HeartbeatStorage: HeartbeatStorageProtocol {
   // MARK: - Instance Management
 
   /// Statically allocated cache of `HeartbeatStorage` instances keyed by string IDs.
-  private static var cachedInstances: [String: WeakContainer<HeartbeatStorage>] = [:]
+  #if compiler(>=6)
+    // In Swift 6, this property is not concurrency-safe because it is
+    // nonisolated global shared mutable state. Because this target's
+    // deployment version does not support Swift concurrency, it is marked as
+    // `nonisolated(unsafe)` to disable concurrency-safety checks. The
+    // property's access is protected by an external synchronization mechanism
+    // (see `instancesLock` property).
+    private nonisolated(unsafe) static var cachedInstances: [
+      String: WeakContainer<HeartbeatStorage>
+    ] =
+      [:]
```

Comment:
nit: I'd move the `[:]` up to the line above (and line 70) if our formatter lets you.

Reply:
nice, thanks! I didn't realize there was a chance I could change what the style script did and it not undo it.